library/cc/engine_builder.cc

```diff
@@ -322,14 +328,21 @@ EngineSharedPtr EngineBuilder::build() {
   } else {
     config_str = config_override_for_tests_;
   }
-  auto envoy_engine =
+  envoy_engine_t envoy_engine =
       init_engine(this->callbacks_->asEnvoyEngineCallbacks(), null_logger, null_tracker);
 
-  for (auto it = key_value_stores_.begin(); it != key_value_stores_.end(); ++it) {
+  for (const auto& [name, store] : key_value_stores_) {
+    // TODO(goaway): This leaks, but it's tied to the life of the engine.
```

Comment:
not sure whether we should keep assigning this to goaway - no strong opinion

Reply:
Done!